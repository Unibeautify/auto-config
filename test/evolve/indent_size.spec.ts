import { testGenetic } from "../testGenetic";
import prettier from "@unibeautify/beautifier-prettier";
import jsBeautify from "@unibeautify/beautifier-js-beautify";

testGenetic("indent_size=2", {
  beautifiers: [prettier, jsBeautify],
  language: "JavaScript",
  originalText: `if (true) { helloWorld(); }`,
  desiredText: `if (true) {\n  helloWorld();\n}`,
});

testGenetic(
  "indent_size=4",
  {
    beautifiers: [prettier, jsBeautify],
    language: "JavaScript",
    originalText: `if (true) { helloWorld(); }`,
    desiredText: `if (true) {\n    helloWorld();\n}`,
  },
  true
);

testGenetic("indent_size=3", {
  beautifiers: [prettier, jsBeautify],
  language: "JavaScript",
  originalText: `if (true) { helloWorld(); }`,
  desiredText: `if (true) {\n   helloWorld();\n}`,
});

testGenetic("indent_size=5", {
  beautifiers: [prettier, jsBeautify],
  language: "JavaScript",
  originalText: `if (true) { helloWorld(); }\nif (true) { helloWorld(); }`,
  desiredText: `if (true) {\n     helloWorld();\n}\nif (true) {\n     helloWorld();\n}\n`,
});
