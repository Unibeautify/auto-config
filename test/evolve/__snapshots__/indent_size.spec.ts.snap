// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`indent_size=0: beautifiedText 1`] = `
if (true) {
helloWorld();
helloWorld();
}
`;

exports[`indent_size=0: entity 1`] = `
Object {
  "options": Object {
    "beautifiers": Array [
      "JS-Beautify",
    ],
    "indent_size": 0,
  },
}
`;

exports[`indent_size=0: fitness 1`] = `0`;

exports[`indent_size=2: beautifiedText 1`] = `
if (true) {
  helloWorld();
}
`;

exports[`indent_size=2: entity 1`] = `
Object {
  "options": Object {
    "beautifiers": Array [
      "JS-Beautify",
    ],
    "indent_size": 2,
  },
}
`;

exports[`indent_size=2: fitness 1`] = `0`;

exports[`indent_size=3: beautifiedText 1`] = `
if (true) {
   helloWorld();
}
`;

exports[`indent_size=3: entity 1`] = `
Object {
  "options": Object {
    "beautifiers": Array [
      "Prettier",
    ],
    "indent_size": 3
  },
}
`;

exports[`indent_size=3: fitness 1`] = `0`;

exports[`indent_size=5: beautifiedText 1`] = `
if (true) {
     helloWorld();
}
if (true) {
     helloWorld();
}

`;

exports[`indent_size=5: entity 1`] = `
Object {
  "options": Object {
    "beautifiers": Array [
      "Prettier",
    ],
    "indent_size": 5,
  },
}
`;

exports[`indent_size=5: fitness 1`] = `0`;
